18/RHOSRFE-61_new_cli_flags

```diff
@@ -2,87 +2,163 @@
 
 ### Problem Descriptions
 
-This is where you should describe the problem that you are trying to solve.
-Be succinct but specific, examples are great, links to bug reports are even
-better. If you don't convince the reader there is a problem, then they are
-unlikely to agree to work with you on a solution.
+This specification defines several new CLI flags and modifies the behavior of the --noop flag
+to improve the os-net-config workflow. These new flags will allow rollback of changes, allow
+os-net-config to continue making modifications even if one interface fails to be configured
+as requested, and will add functionality to look for IP address conflicts in the inteface
+configurations which have static IP addresses.
 
 ### Background Context (Optional)
 
-If there is some more background context, provide it here, otherwise feel
-free to delete this section and move on to the proposed solution.
+Currently os-net-config does not have the ability to rollback if an inteface configuration
+leads to an error, instead the execution will halt and typically the network configuration will
+be left in a state with broken or limited connectivity. This situation is worse with the ifcfg
+provider than with the nmstate provider, because nmstate supports rollback natively. Furthermore,
+IP address conflict detection only happens after the interface configurations are applied. Since
+the interface configurations are applied serially, when an error occurs further interfaces may
+not be brought up, leading to inconsistent or broken state.
 
 ### Proposed Changes
 
-This is a summary of the proposed solutions.
+Several new CLI flags would be added in order to support rollbacks, continuing to apply interface
+configurations after an error occurs, and new funcionality to check for IP address conflicts when
+the --noop flag is used would be added.
 
 #### List of Proposed Changes
 
-1. Solution 1
-2. Solution 2
+1. Add the --current-config flag which would be used to pass a copy of the config template which
+   would be used to rollback to if the --rollback flag is used, or possibly to modify the behavior
+   of the --cleanup flag to limit cleanup to only removing interface configurations which were
+   present in the --current-config configuration provided but are not present in the new config
+   template which is being applied.
+2. Add the --rollback flag which would either support rolling back to the original configuration
```

Comment:
Instead of a complete rollback, can we consider configuring the minimal networks needed to enable ssh and other bare minimum tasks ?